vignettes/data-preparation.Rmd

```diff
@@ -0,0 +1,399 @@
+
+
+```{r local-deployment, eval = FALSE}
+# Create the app data structure
+run_gsm_app(
+  dfAnalyticsInput = dfAnalyticsInput,
+  dfBounds = dfBounds,
+  dfGroups = dfGroups,
+  dfMetrics = dfMetrics,
+  dfResults = dfResults,
+  fnFetchData = fnFetchData,
+  fnCountData = ConstructDataCounter(fnFetchData),
+  chrDomains = c(AE = "Adverse Events", DATACHG = "Data Changes", DATAENT = "Data Entry",
+    ENROLL = "Enrollment", LB = "Lab", PD = "Protocol Deviations", QUERY = "Queries",
+    STUDCOMP = "Study Completion", SUBJ = "Subject Metadata", SDRGCOMP =
+    "Treatment Completion"),
+  lPlugins = NULL,
+  strTitle = ExtractAppTitle(dfGroups),
+  strFavicon = "angles-up",
+  strFaviconColor = "#FF5859",
+  tagListExtra = NULL,
+  fnServer = NULL
+)
+```
+
+# Troubleshooting
+
+## Common Issues
+
+1.  **Data Validation Errors**: Ensure all required columns are present and correctly named
+2.  **Missing MetricID Links**: Verify MetricID values are consistent across all data frames
+3.  **Date Format Issues**: Ensure all date columns are properly formatted as Date objects
+4.  **Memory Issues**: For large datasets, consider data filtering and optimization
+5.  **Fetch Function Errors**: Test your `fnFetchData` function with various parameter combinations
+
+
+# Summary
+
+This vignette has demonstrated a comprehensive workflow for preparing clinical trial data for use with {gsm.app} using the {gsm.core} and {gsm.reporting} packages. The approach provides:
+
+## Key Benefits
+
+1.  **Validated Statistical Methods**: Uses peer-reviewed KRI calculation methods from {gsm.core}
+2.  **Standardized Data Model**: Ensures compatibility across the {gsm} ecosystem
+3.  **Professional Reporting**: Produces publication-ready visualizations and tables
+4.  **Automated Workflows**: Reduces manual intervention and potential errors
+5.  **Quality Assurance**: Built-in validation and data quality checks
+6.  **Scalable Architecture**: Suitable for both development and production environments
+
+## Next Steps
+
+-   **Customize Workflows**: Modify YAML configuration files to match your specific study requirements
+-   **Validate Results**: Use the validation functions provided in {gsm.core} to verify data quality
+-   **Deploy Applications**: Choose the deployment method that best fits your infrastructure needs
+-   **Monitor Performance**: Implement the performance optimization techniques for larger datasets
+
+## Additional Resources
+
+-   [gsm.core Documentation](https://gilead-biostats.github.io/gsm.core/)
+-   [gsm.reporting Documentation](https://gilead-biostats.github.io/gsm.reporting/)
+-   [gsm.app Repository](https://github.com/Gilead-BioStats/gsm.app)
+
+For questions or issues, please refer to the package documentation or submit issues to the appropriate GitHub repositories.
```

Comment:
This feels generic to the gsm family of packages. I don't think it belongs in this vignette.